app/src/userpages/components/ProfilePage/APICredentials/CredentialsControl/index.jsx

```diff
@@ -30,31 +37,55 @@ export default class CredentialsControl extends Component<Props> {
     }
 
     render() {
+        const { streamId, editStreamResourceKey, editMyResourceKey, showPermissionType } = this.props
+
         return (
             <div>
                 <div className={styles.keyList}>
-                    {this.props.keys.map((key: ResourceKey) => (
-                        <KeyField
-                            className={styles.singleKey}
-                            key={key.id}
-                            keyName={key.name}
-                            value={key.id}
-                            hideValue
-                            allowEdit
-                            onSave={() => {
-                                alert('Editing is not supported yet!') // eslint-disable-line no-alert
-                                return Promise.resolve()
-                            }}
-                            allowDelete
-                            disableDelete={this.props.disableDelete}
-                            onDelete={() => {
-                                if (key.id) {
-                                    this.props.removeKey(key.id)
+                    {this.props.keys.map((key: ResourceKey, index: number) => (
+                        <Fragment key={key.id}>
+                            {!index && (
+                                <Row>
+                                    <Col md={12} lg={11} />
+                                    <Col md={12} lg={1}>
+                                        {showPermissionType && (
+                                            <Translate
+                                                value="userpages.streams.edit.configure.permission"
+                                                className={styles.permissionColHeading}
+                                            />
+                                        )}
+                                    </Col>
+                                </Row>
+                            )}
+                            <KeyField
+                                className={cx(styles.singleKey, {
+                                    [styles.firstKey]: showPermissionType && !index,
+                                })}
+                                key={key.id}
+                                keyName={key.name}
+                                value={key.id}
+                                hideValue
+                                allowEdit
+                                onSave={(keyName, value, keyPermission) => {
+                                    if (streamId && key.id && keyName && keyPermission && editStreamResourceKey) {
+                                        editStreamResourceKey(streamId, key.id, keyName, keyPermission)
```

Comment:
Need to return these promises so that the key editor knows to wait for the operation to complete.